Public/Src/Engine/Scheduler/Graph/PipGraph.MacOsDefaults.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        /// <nodoc />
+        public class MacOsDefaults
+        {
+            private static readonly SortedReadOnlyArray<FileArtifact, OrdinalFileArtifactComparer> s_emptySealContents
+                = CollectionUtilities.EmptySortedReadOnlyArray<FileArtifact, OrdinalFileArtifactComparer>(OrdinalFileArtifactComparer.Instance);
+
+            private readonly PipProvenance m_provenance;
+            private readonly FileArtifact[] m_inputFiles;
+            private readonly DirectoryArtifact[] m_inputDirectories;
+            private readonly FileArtifact[] m_untrackedFiles;
+            private readonly DirectoryArtifact[] m_untrackedDirectories;
+
+            /// <nodoc />
+            public MacOsDefaults(PathTable pathTable, PipGraph.Builder pipGraph)
+            {
+                m_provenance = new PipProvenance(
+                    0,
+                    ModuleId.Invalid,
+                    StringId.Invalid,
+                    FullSymbol.Invalid,
+                    LocationData.Invalid,
+                    QualifierId.Unqualified,
+                    PipData.Invalid);
+
+                var user = Environment.GetEnvironmentVariable("USER") ?? string.Empty;
+
+                m_inputFiles =
+                    new[]
+                    {
+                        FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, MacPaths.Etc)),
+                        FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, MacPaths.TmpDir)),
+                    };
+
+                // Sealed Source inputs
+                m_inputDirectories =
+                    new[]
+                    {
+                        GetSourceSeal(pathTable, pipGraph, MacPaths.Applications),
+                        GetSourceSeal(pathTable, pipGraph, MacPaths.UsrBin),
+                        GetSourceSeal(pathTable, pipGraph, MacPaths.UsrInclude),
+                        GetSourceSeal(pathTable, pipGraph, MacPaths.UsrLib),
+                        GetSourceSeal(pathTable, pipGraph, MacPaths.Library),
+                        GetSourceSeal(pathTable, pipGraph, $"/Users/{user}/Library/MobileDevice/Provisioning Profiles"),
+                    };
+
+                m_untrackedFiles =
+                    new[]
+                    {
+                        // login.keychain is created by the OS the first time any process invokes an OS API that references the keychain.
+                        // Untracked because build state will not be stored there and code signing will fail if required certs are in the keychain
+                        FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, $"/Users/{user}/Library/Keychains/login.keychain-db")),
+                        FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, $"/Users/{user}/Library/Keychains/login.keychain")),
+                        FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, $"/Users/{user}/.CFUserTextEncoding")),

```

Comment:
consider moving all these paths to `MacPaths`.

Reply:
had to make 'm static readly due to user folder dependency. but I moved it.

---
In reply to: [274229286](https://github.com/Microsoft/BuildXL/pull/125#discussion_r274229286) [](ancestors = 274229286)